lib/src/realm_class.dart

```diff
@@ -19,231 +19,79 @@
-import 'realm_bindings_win.dart';
+import 'native/realm_core.dart';
 
 export 'collection.dart';
 export 'list.dart';
 export 'results.dart';
 export 'realm_object.dart';
 export "configuration.dart";
 export 'realm_property.dart';
-export 'dynamic_object.dart';
 export 'helpers.dart';
 
-NativeLibrary? RealmLib;
-void setRealmLib(DynamicLibrary realmLibrary) {
-  RealmLib = NativeLibrary(realmLibrary);
-}
-
-/// The callback type to use with `Realm.write`
-typedef void VoidCallback();
-
-/// The callback type to use with `Realm.addListener`
-typedef void ListenerCallback(dynamic sender, String event);
-
-void _inspect(dynamic arg1, dynamic arg2, dynamic arg3, dynamic arg4, dynamic arg5) {
-   Object k;
-}
-
-/// An event type used when adding event listeners with `Realm.addListener`
-enum Event {
-  /// The callback will be called when objects in the `Realm` changed
-  change,
-
-  /// The callback will be called when the `Realm` schema changes
-  schema
-}
+void setRealmLib(DynamicLibrary realmLibrary) => setRealmLibrary(realmLibrary);
 
 /// A Realm instance represents a Realm database.
-class Realm extends DynamicObject {
-  // Used from native code
-  Realm._constructor();
-
-  /// Opens a Realm using a [Configuration] object
-  factory Realm(Configuration config) native "Realm_constructor";
-
-  //native code expects first argument this instance. For static methods pass null
-  
-  /// Get the schemaVersion version of the Realm at [path].
-  /// 
-  /// For details about the schema version see [Configuration.schemaVersion]
-  static double schemaVersion(String path) {
-    return _schemaVersion(null, path);
-  }
-  static double _schemaVersion(Object? nullptr, String path) native "Realm_schemaVersion";
-
-  /// Returns `true` if the Realm already exists on [path].
-  static bool exists(String path) {
-    return _exists(null, new Configuration()..path = path);
-  }
-  static bool _exists(Object? nullptr, Configuration config) native "Realm_exists";
-
-  /// DO NOT USE.
-  static bool clearTestState() {
-    // TODO: fix clearTestState with FFI
-    //return _clearTestState(null);
-    return true;
-  }
-  //move this to the tests and make it an extensions method that calls clearTestState and passes the correct `this`
-  static bool _clearTestState(Object nullptr) native "Realm_clearTestState";
-
-  /// Delete the Realm file at [path]
-  static void deleteFile(String path) {
-    File realmFile = new File(path);
-    if (!realmFile.existsSync()) {
-      throw new RealmException("The realm file does not exists at path $path");
+class Realm {
+  /// The [Configuration] object of this [Realm]
+  Configuration get config => throw RealmException("not implemented");
+  late RealmHandle _realm;
+  late final _Scheduler _scheduler;
+
+  /// Opens a Realm using the default or a custom [Configuration] object
+  Realm(Configuration config) {
+    if (!config.schema.isEmpty) {
+      realmCore.validateSchema(config.schema);
```

Comment:
Why do we need this? I imagine we'll validate the schema when we try to open the Realm anyway.

Reply:
yes this is something we discussed already internally . I will be removing this validation

Reply:
done